jina/peapods/pods/__init__.py

```diff
@@ -150,20 +144,12 @@ def _parse_base_pod_args(
             args.scheduling = SchedulerType.ROUND_ROBIN
             if getattr(args, 'uses_before', None):
                 self.is_head_router = True
-                parsed_args['head'] = self._copy_to_head_args(
-                    args, polling_type.is_push
-                )
+                parsed_args['head'] = self._copy_to_head_args(args, polling_type)
             if getattr(args, 'uses_after', None):
                 self.is_tail_router = True
-                parsed_args['tail'] = self._copy_to_tail_args(
-                    args, polling_type.is_push
-                )
-            parsed_args[attribute] = self._set_peas_args(
+                parsed_args['tail'] = self._copy_to_tail_args(args, polling_type)
+            parsed_args[attribute] = self._set_middle_args(
```

Comment:
do not change the name yet, I would like to see what is needed here to share between Pea and Pod

Reply:
I thought `_set_middle_args` would be a good method name for everything which is between head and tail.
This method is then be used in the compound pod and in the pod.
In the compound pod the middle args are for the replicas and in the pod the middle args are as we used to for the peas.
I can also change it back if you want.

Reply:
since it is not in the common interface and is private I think it is better to have self explanatory

Reply:
It is actually the same function_name used for pods and compound pods.

Reply:
I think we will refactor this part anyways in the close future.

Reply:
but not declared in interface, different names make the usage description explicit

Reply:
okay I will do it

Reply:
I added the base method as abstract method.

Reply:
Private methods should not be part of the interface. I think specialized names are better. Set peas args for Pod and set pod args for CompoundPod would be the more explainable ones.

After all, you are seeing that any BasePod needs to implement this method, which no external component should care about

Reply:
okay, changed thanks